server_cmds.go

```diff
@@ -351,19 +365,48 @@ func (s *ServerMethods) PrivilegeKeyAdd(ttype, id1, id2 int, options ...CmdArg)
 
 // OnlineClient represents a client online on a virtual server.
 type OnlineClient struct {
-	ID          int    `ms:"clid"`
-	ChannelID   int    `ms:"cid"`
-	DatabaseID  int    `ms:"client_database_id"`
-	Nickname    string `ms:"client_nickname"`
-	Type        int    `ms:"client_type"`
-	Away        bool   `ms:"client_away"`
-	AwayMessage string `ms:"client_away_message"`
+	// Following variables are always returned by ClientList().
+	ID         int    `ms:"clid"`
+	ChannelID  int    `ms:"cid"`
+	DatabaseID int    `ms:"client_database_id"`
+	Nickname   string `ms:"client_nickname"`
+	Type       int    `ms:"client_type"`
+	// Following variables are optional and can be requested in ClientList() to get extended client information.
```

Comment:
Would it make sense to split each of these out into separate embedded structs such that they correspond to the command line flags, that way the caller can use minimal dereferencing when consuming?

Thoughts?

Reply:
Sounds like a good idea to me. I tried implementing this in the new commit and personally think it *looks* better. I'm a little too inexperienced to know if this would fulfil the goal of ensuring 'the caller can use minimal dereferencing when consuming'.  I don't think I fully understand what this means yet. Eager to read what your thoughts are on my attempt.